lib/table/normalize.js

```diff
@@ -0,0 +1,198 @@
+  // Normalize config
+  let index = this.document._tableIndex++;
+  this._id = new String(this.opts.id ?? `table-${index}`);
+  this._position = {
+    x: this.document.sizeToPoint(this.opts.position?.x, this.document.x),
+    y: this.document.sizeToPoint(this.opts.position?.y, this.document.y),
+  };
+  this._maxWidth = this.document.sizeToPoint(
+    this.opts.maxWidth,
+    this.document.page.width -
+      this.document.page.margins.right -
+      this._position.x,
+  );
+
+  const { defaultStyle, defaultColStyle, defaultRowStyle } =
+    normalizedDefaultStyle(this.opts.defaultStyle);
+  this._defaultStyle = defaultStyle;
+
+  let colStyle;
+  if (this.opts.columnStyles) {
+    if (Array.isArray(this.opts.columnStyles)) {
+      colStyle = (i) => this.opts.columnStyles[i];
+    } else if (typeof this.opts.columnStyles === 'function') {
+      colStyle = this.opts.columnStyles;
+    } else if (typeof this.opts.columnStyles === 'object') {
+      colStyle = () => this.opts.columnStyles;
+    }
+  }
+  if (!colStyle) colStyle = () => ({});
+  this._colStyle = normalizedColumnStyle.bind(this, defaultColStyle, colStyle);
+
+  let rowStyle;
+  if (this.opts.rowStyles) {
+    if (Array.isArray(this.opts.rowStyles)) {
+      rowStyle = (i) => this.opts.rowStyles[i];
+    } else if (typeof this.opts.rowStyles === 'function') {
+      rowStyle = this.opts.rowStyles;
+    } else if (typeof this.opts.rowStyles === 'object') {
+      rowStyle = () => this.opts.rowStyles;
+    }
+  }
+  if (!rowStyle) rowStyle = () => ({});
+  this._rowStyle = normalizedRowStyle.bind(this, defaultRowStyle, rowStyle);
+}
+
+/**
+ * Convert text into a string
+ * - null and undefined are preserved (as they will be ignored)
+ * - everything else is run through `String()`
+ *
+ * @param {*} text
+ * @returns {string}
+ * @private
+ */
+export function normalizeText(text) {
+  // Parse out text
+  if (text !== null && text !== undefined) text = String(text);
+  return text;
+}
+
+/**
+ * Normalize a cell config
+ *
+ * @this PDFTable
+ * @memberOf PDFTable
+ * @param {TableCellStyle} cell - The cell to mutate
+ * @param {number} rowIndex - The cells row
+ * @param {number} colIndex - The cells column
+ * @returns {NormalizedTableCellStyle}
+ * @private
+ */
+export function normalizeCell(cell, rowIndex, colIndex) {
+  const colStyle = this._colStyle(colIndex);
+  let rowStyle = this._rowStyle(rowIndex);
+
+  const font = deepMerge({}, colStyle.font, rowStyle.font, cell.font);
+
+  // Initialize cell context
+  const rollbackFont = this.document._fontSource;
+  const rollbackFontSize = this.document._fontSize;
+  const rollbackFontFamily = this.document._fontFamily;
+  if (font.src) this.document.font(font.src, font.family);
+  if (font.size) this.document.fontSize(font.size);
+
+  // Refetch rowStyle to reflect font changes
+  rowStyle = this._rowStyle(rowIndex);
+
+  COLUMN_FIELDS.forEach((f) => delete colStyle[f]);
+  ROW_FIELDS.forEach((f) => delete rowStyle[f]);
```

Comment:
Why these?

Reply:
just cleanup because those fields are only used for sizing of a cell not the actual cell contents

Reply:
If there's no functional reason, this code should be removed.

Reply:
it could be argued that it reduces memory since some of these cells may be stored in the rowSpan buffer? but other than that it serves no purpose

Reply:
delete changes the shape of object in memory leading to performance overhead

So i still think it should be removed

Reply:
just tested, turns out you are right, plus since these are all primitives its such a small memory impact, so ive removed in latest commit